### PR TITLE
Add support for building w/ Swift Package Manager

### PR DESCRIPTION
**Pull request by gfontenot**
This does _not_ add support for running `swift test`, unfortunately. The
test dependency stuff seems to not be fully flushed out in the current
build and AFAIK, there's no way to point to a specific branch, only a tag.
So we can't actually point to the proper version of SwiftCheck right now
anyway.

### CODE REVIEW

File: Runes.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -701,8 +696,12 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				DEBUG_INFORMATION_FORMAT = dwarf;
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(PROJECT_DIR)/Carthage/Build/tvOS",
```
Comment (tonyd256) [inline Runes.xcodeproj/project.pbxproj:701]: should this be in here if this is for SPM?

File: Runes.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -715,8 +714,12 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(PROJECT_DIR)/Carthage/Build/tvOS",
```
Comment (tonyd256) [inline Runes.xcodeproj/project.pbxproj:719]: same

**Comment by tonyd256** [comment]
Looks like there are some Carthage references in the project file besides the ones I commented on... should those be purged?

**Comment by gfontenot** [comment]
@tonyd256 SPM doesn't use the project file at all, it's completely ignored. So those changed won't matter for SPM. That was me trying to get the project working correctly again on its own.

**Comment by tonyd256** [comment]
👍

**Pull request merged**